reanalysis/validate_categories.py

```diff
@@ -207,19 +220,16 @@ def clean_initial_results(
             f'{support_id}'
         )
 
-        # create a section for this sample if it doesn't exist
-        if each_instance.sample not in clean_results:
-            clean_results[each_instance.sample] = {}
-
         # get an existing object, or use the current one
-        variant_object = clean_results.setdefault(each_instance.sample, {}).setdefault(
+        variant_object = clean_results[each_instance.sample].setdefault(
             var_uid, each_instance
         )
 
         # combine any possible reasons, and add
         clean_results[each_instance.sample][
```

Comment:
Can this be simplified to:
`variant_object.reasons.update(each_instance.reasons)`

Reply:
Aye, much cleaner!

Reply:
Changed this whole section to something simpler - if something is already at that index, update. If it's empty, populate